pkg/operator/operator_config.go

```diff
@@ -240,6 +238,30 @@ func (r *operatorConfigReconciler) Reconcile(ctx context.Context, req reconcile.
 	return reconcile.Result{}, nil
 }
 
+func (r *operatorConfigReconciler) ensureOperatorConfig(ctx context.Context, logger logr.Logger, req reconcile.Request) (*monitoringv1.OperatorConfig, error) {
+	exists := true
+	config := &monitoringv1.OperatorConfig{}
+	if err := r.client.Get(ctx, req.NamespacedName, config); apierrors.IsNotFound(err) {
+		logger.Info("no operatorconfig created yet")
+		exists = false
+	} else if err != nil {
+		return nil, fmt.Errorf("get operatorconfig for incoming: %q: %w", req.String(), err)
+	}
+	defaulter := &operatorConfigDefaulter{
+		projectID: r.opts.ProjectID,
+		location:  r.opts.Location,
+		cluster:   r.opts.Cluster,
+	}
+	configUpdate := config.DeepCopy()
+	defaulter.update(configUpdate)
+	if exists && !reflect.DeepEqual(config, configUpdate) {
```

Comment:
suggestion (nit):

I'm wondering, given the relative infrequent updates to the OperatorConfig, it would be cheaper to just update it every time either way to save on deep copies and equal evaluations, i.e.
```go
// ...
defaulter.update(config)

// Only try and update the OperatorConfig with defaults if it exists.
if exists && err := r.client.Update(ctx, config); err != nil {
		return nil, fmt.Errorf("default operatorconfig: %w", err)
}

return config
```

Reply:
Made an edit - my formatting got messed up - sorry.